checker/src/main/java/org/checkerframework/checker/index/upperbound/OffsetEquation.java

```diff
@@ -266,6 +271,16 @@ private Integer evalConstantTerm(Receiver termReceiver) {
                     }
                 }
             }
+        } else if (factory != null && termReceiver instanceof FlowExpressions.LocalVariable) {
+            AnnotationMirror am =
+                    ((BaseAnnotatedTypeFactory) factory)
+                            .getTypeFactoryOfSubchecker(ValueChecker.class)
+                            .getAnnotatedType(
+                                    ((FlowExpressions.LocalVariable) termReceiver).getElement())
+                            .getAnnotation(IntVal.class);
+            if (am != null && ValueAnnotatedTypeFactory.getIntValues(am).size() == 1) {
+                return ValueAnnotatedTypeFactory.getIntValues(am).get(0).intValue();
```

Comment:
You should create a version of `IndexUtil#getExactValue` that takes an element and call that method.